src/site/content/en/blog/bfcache/index.md

```diff
@@ -491,13 +491,59 @@ should be similar for other analytics tools:
 gtag('event', 'page_view');
 
 window.addEventListener('pageshow', (event) => {
-  if (event.persisted === true) {
-    // Send another pageview if the page is restored from bfcache.
+  // Send another pageview if the page is restored from bfcache.
+  if (event.persisted) {
+    gtag('event', 'page_view');
+  }
+});
+```
+
+### Measuring your bfcache hit ratio
+
+As well as measuring pageview measurements for bfcache page navigations, you
+may wish to also track whether the bfcache was used to help identify pages
+that are not utilising the bfcache. For example:
+
+```js
+// Send a pageview when the page is first loaded.
+gtag('event', 'page_view');
+
+window.addEventListener('pageshow', (event) => {
+
+  // Send another pageview if the page is restored from bfcache.
+  if (event.persisted) {
     gtag('event', 'page_view');
   }
+
+  // For Back/Forward navigations, send track bfcache usage.
+  if (performance.getEntriesByType('navigation')[0].type == "back_forward") {
```

Comment:
There needs to be an `else` here because a page that was previously of type `back_forward` could later be restored from bfcache, and you wouldn't want to send two events.

Reply:
I'm sending a `page_view` event in the first instance (only for restores for tracking the page load), and a generic event in the second instance (for tracking bfcache usage).

I agree that is two events for bfcache restores, but I did look into whether you could add `event_label`, `event_category` and `value` on a `page_view` event to combine them into one event, but didn't seem to the case - even though it's still an "event".

Can give it another go if you think this should be possible?

Reply:
Sure, tracking this as a dimension is [how the web-vitals JS library does it] and probably makes sense here as well.

In that case, you'd want to make sure that every `page_view` event has a `navigation_type` (or whatever name you want) dimension set, and in cases of a bfcache restore you would manually set `bfcache` or `back_forward_cache`.

Reply:
Yeah I thought about custom dimensions, but the relatively low limit on those (at least in old school GA - not looked at GA4) means I try to avoid over using them. Plus the explanation of setting those up versus the simplicity here for this simple code snippet puts me off.

Reply:
FYI, GA4 doesn't necessarily require any custom dimension set (see: [web.dev/vitals-ga4](https://web.dev/vitals-ga4/)), and given that GA3 is deprecated, I think it makes sense to assume GA4 going forward.

Also, we don't necessarily have to make these GA examples, for simplicity we could update them to be generic functions, e.g.:

```js
sendToAnalytics({
  eventName: 'page_view',
  navigationType: 'back_forward_cache',
});

Reply:
So that becomes something like this?

```js
// Send a pageview when the page is first loaded, including the navigation type
gtag('event', 'page_view',
  { dimensionX: performance.getEntriesByType('navigation')[0].type }
);

window.addEventListener('pageshow', (event) => {

  // Send another pageview if the page is restored from bfcache.
  if (event.persisted) {
    gtag('event', 'page_view',
      { dimensionX: "back_forward_cache" }
    );
  }

});
```

Then we get the following events in GA:
- `navigation`
- `reload`
- `back_forward`
- `back_forward_cache`
- ...etc.

It works, but then we need to explain to the user to measure it they need to total `back_forward` and `back_forward_cache` and divide by `back_forward_cache` to get the bfcache hit ratio, ignoring any non-baack/force navigations.

It does save one event I admit, but I think the current code and reporting of it is just a heck of a lot easy for readers to understand.

Plus with the current code it's easy to see this ratio in the standard event report in GA:

<img width="940" alt="image" src="https://user-images.githubusercontent.com/10931297/176952000-193d286f-4ddc-4c07-ae92-7d8726ae9fc0.png">

> Sure, tracking this as a dimension is [how the web-vitals JS library does it] and probably makes sense here as well.

Well that's not really true. The web vitals JS library still [sends a "Web Vitals" event](https://github.com/GoogleChrome/web-vitals#using-gtagjs-universal-analytics) (in fact one per CWV so multiple events for each page load) - it just adds the optional debugging information in the dimensions. The actual values are sent in the event `value` field, exactly like I'm doing with the current code.

Reply:
Discussed offline and updated.